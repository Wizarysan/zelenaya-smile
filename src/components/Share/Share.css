@media screen and (max-width: 767px) {
  .mobile-hide {
    display: none; }
  .mobile-show {
    display: initial; } }

@media screen and (min-width: 768px) {
  .mobile-hide {
    display: initial; }
  .mobile-show {
    display: none; } }

.animation__photo-enter {
  filter: blur(2px) grayscale(100%); }

.animation__photo-enter-active.animation__photo-enter {
  filter: blur(0px) grayscale(0%);
  transition: all 200ms ease-in; }

.animation__photo-leave {
  opacity: 1; }
  .animation__photo-leave-active.animation__photo-leave {
    opacity: 0.01;
    transition: opacity 200ms ease-in; }

@keyframes bump {
  0% {
    transform: scale(1); }
  50% {
    transform: scale(1.6); }
  100% {
    transform: scale(1); } }

.animation__bump {
  animation: bump .5s ease-out; }

.share {
  padding-top: 5px;
  padding-bottom: 30px; }
  .share__button {
    display: inline-block;
    margin-right: 5px;
    outline: none;
    transition: all .3s; }
    .share__button div {
      display: inline-block; }
    .share__button:hover {
      filter: brightness(120%); }
  .share__icon {
    display: inline-block;
    vertical-align: middle; }
  .share__count {
    display: inline-block;
    padding-right: 5px;
    vertical-align: middle;
    color: white; }
