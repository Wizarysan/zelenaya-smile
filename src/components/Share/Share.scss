@import './../../styles/shared.scss';

.share {
  padding-top: 5px;
  padding-bottom: 30px;
  &__button {
    display: inline-block;
    margin-right: 5px;
    outline: none;
    transition: all .3s;
    div {
      display: inline-block;
    }
    &:hover {
      filter: brightness(120%);
    }
  }
  &__icon {
    display: inline-block;
    vertical-align: middle;
  }
  &__count {
    display: inline-block;
    padding-right: 5px;
    vertical-align: middle;
    color: white;
  }
}
