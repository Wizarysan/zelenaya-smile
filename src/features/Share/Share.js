import React from 'react';
import {Link} from 'react-router';
import {Helmet} from "react-helmet";
import './Share.css';
import {
  ShareButtons,
  ShareCounts,
  generateShareIcon
} from 'react-share';
const {
  FacebookShareButton,
  TwitterShareButton,
  VKShareButton,
  OKShareButton,
} = ShareButtons;
const {
  FacebookShareCount,
  VKShareCount,
  OKShareCount,
} = ShareCounts;
const VKIcon = generateShareIcon('vk');
const FacebookIcon = generateShareIcon('facebook');
const OKIcon = generateShareIcon('ok');
const TwitterIcon = generateShareIcon('twitter');

class Share extends React.Component {
  render() {
    let shareUrl = window.location.href;
    let shareImg = this.props.image;

    //TODO refactor shareCount to one function
    return (
      <div className="share">
        <Helmet>
          <meta property="og:type" content="article" />
          <meta property="og:url" content={shareUrl} />
          <meta property="og:image" content={shareImg} />
        </Helmet>
        <VKShareButton style={{backgroundColor:'#45668e'}} className="share__button"
          url={shareUrl} title={`Улыбнись :) - Зелёная Точка. ${document.title}`}
          image={shareImg}>
          <VKIcon className="share__icon" size={this.props.size}  />
          <VKShareCount url={shareUrl}>
            {shareCount => {
              if (shareCount>0) return <span className="share__count">{shareCount}</span>
            }}
          </VKShareCount>
        </VKShareButton>
        <FacebookShareButton style={{backgroundColor:'#3b5998'}}  className="share__button"
          url={shareUrl} quote={`Улыбнись :) - Зелёная Точка. ${document.title}`}>
          <FacebookIcon className="share__icon" size={this.props.size}  />
          <FacebookShareCount url={shareUrl}>
            {shareCount => {
              if (shareCount>0) return <span className="share__count">{shareCount}</span>
            }}
          </FacebookShareCount>
        </FacebookShareButton>
        <OKShareButton style={{backgroundColor:'#f2720c'}} className="share__button"
          url={shareUrl} title={`Улыбнись :) - Зелёная Точка. ${document.title}`}
          picture={shareImg}>
          <OKIcon className="share__icon" size={this.props.size}  />
          <OKShareCount url={shareUrl}>
            {shareCount => {
              if (shareCount>0) return <span className="share__count">{shareCount}</span>
            }}
          </OKShareCount>
        </OKShareButton>
        <TwitterShareButton className="share__button"
          url={shareUrl} title={`Улыбнись :) - Зелёная Точка. ${document.title}`}>
          <TwitterIcon className="share__icon" size={this.props.size}  />
        </TwitterShareButton>
      </div>
    );
  }
}

export default Share;
