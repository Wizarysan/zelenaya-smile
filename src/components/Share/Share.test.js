import React from 'react';
import Share from './Share.js';
import renderer from 'react-test-renderer';

it('renders correctly with test props', () => {
  const tree = renderer
    .create(<Share size="64" image="test.jpg"/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});