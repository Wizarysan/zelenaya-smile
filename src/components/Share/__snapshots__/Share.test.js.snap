exports[`test renders correctly with test props 1`] = `
<div
  className="share">
  <div
    className="SocialMediaShareButton SocialMediaShareButton--vk share__button"
    onClick={[Function]}
    onKeyPress={[Function]}
    role="button"
    style={
      Object {
        "backgroundColor": "#45668e",
      }
    }
    tabIndex="0">
    <div
      style={
        Object {
          "height": "64",
          "width": "64",
        }
      }>
      <svg
        className="social-icon social-icon--vk share__icon"
        height="64"
        viewBox="0 0 64 64"
        width="64">
        <g>
          <rect
            fill="#45668e"
            height="64"
            style={Object {}}
            width="64" />
        </g>
        <g>
          <path
            d="M44.94,44.84h-0.2c-2.17-.36-3.66-1.92-4.92-3.37C39.1,40.66,38,38.81,36.7,39c-1.85.3-.93,3.52-1.71,4.9-0.62,1.11-3.29.91-5.12,0.71-5.79-.62-8.75-3.77-11.35-7.14A64.13,64.13,0,0,1,11.6,26a10.59,10.59,0,0,1-1.51-4.49C11,20.7,12.56,21,14.11,21c1.31,0,3.36-.29,4.32.2C19,21.46,19.57,23,20,24a37.18,37.18,0,0,0,3.31,5.82c0.56,0.81,1.41,2.35,2.41,2.14s1.06-2.63,1.1-4.18c0-1.77,0-4-.5-4.9S25,22,24.15,21.47c0.73-1.49,2.72-1.63,5.12-1.63,2,0,4.84-.23,5.62,1.12s0.25,3.85.2,5.71c-0.06,2.09-.41,4.25,1,5.21,1.09-.12,1.68-1.2,2.31-2A28,28,0,0,0,41.72,24c0.44-1,.91-2.65,1.71-3,1.21-.47,3.15-0.1,4.92-0.1,1.46,0,4.05-.41,4.52.61,0.39,0.85-.75,3-1.1,3.57a61.88,61.88,0,0,1-4.12,5.61c-0.58.78-1.78,2-1.71,3.27,0.05,0.94,1,1.67,1.71,2.35a33.12,33.12,0,0,1,3.92,4.18c0.47,0.62,1.5,2,1.4,2.76C52.66,45.81,46.88,44.24,44.94,44.84Z"
            fill="white" />
        </g>
      </svg>
    </div>
    <div
      className="SocialMediaShareCount" />
  </div>
  <div
    className="SocialMediaShareButton SocialMediaShareButton--facebook share__button"
    onClick={[Function]}
    onKeyPress={[Function]}
    role="button"
    style={
      Object {
        "backgroundColor": "#3b5998",
      }
    }
    tabIndex="0">
    <div
      style={
        Object {
          "height": "64",
          "width": "64",
        }
      }>
      <svg
        className="social-icon social-icon--facebook share__icon"
        height="64"
        viewBox="0 0 64 64"
        width="64">
        <g>
          <rect
            fill="#3b5998"
            height="64"
            style={Object {}}
            width="64" />
        </g>
        <g>
          <path
            d="M34.1,47V33.3h4.6l0.7-5.3h-5.3v-3.4c0-1.5,0.4-2.6,2.6-2.6l2.8,0v-4.8c-0.5-0.1-2.2-0.2-4.1-0.2 c-4.1,0-6.9,2.5-6.9,7V28H24v5.3h4.6V47H34.1z"
            fill="white" />
        </g>
      </svg>
    </div>
    <div
      className="SocialMediaShareCount" />
  </div>
  <div
    className="SocialMediaShareButton SocialMediaShareButton--ok share__button"
    onClick={[Function]}
    onKeyPress={[Function]}
    role="button"
    style={
      Object {
        "backgroundColor": "#f2720c",
      }
    }
    tabIndex="0">
    <div
      style={
        Object {
          "height": "64",
          "width": "64",
        }
      }>
      <svg
        className="social-icon social-icon--ok share__icon"
        height="64"
        viewBox="0 0 64 64"
        width="64">
        <g>
          <rect
            fill="#f2720c"
            height="64"
            style={Object {}}
            width="64" />
        </g>
        <g>
          <path
            d="M39,30c-1,0-3,2-7,2s-6-2-7-2c-1.1,0-2,0.9-2,2c0,1,0.6,1.5,1,1.7c1.2,0.7,5,2.3,5,2.3l-4.3,5.4   c0,0-0.8,0.9-0.8,1.6c0,1.1,0.9,2,2,2c1,0,1.5-0.7,1.5-0.7S32,39,32,39c0,0,4.5,5.3,4.5,5.3S37,45,38,45c1.1,0,2-0.9,2-2   c0-0.6-0.8-1.6-0.8-1.6L35,36c0,0,3.8-1.6,5-2.3c0.4-0.3,1-0.7,1-1.7C41,30.9,40.1,30,39,30z M32,15c-3.9,0-7,3.1-7,7s3.1,7,7,7c3.9,0,7-3.1,7-7S35.9,15,32,15z M32,25.5   c-1.9,0-3.5-1.6-3.5-3.5c0-1.9,1.6-3.5,3.5-3.5c1.9,0,3.5,1.6,3.5,3.5C35.5,23.9,33.9,22.5,35,22.5z "
            fill="white" />
        </g>
      </svg>
    </div>
    <div
      className="SocialMediaShareCount" />
  </div>
  <div
    className="SocialMediaShareButton SocialMediaShareButton--twitter share__button"
    onClick={[Function]}
    onKeyPress={[Function]}
    role="button"
    style={Object {}}
    tabIndex="0">
    <div
      style={
        Object {
          "height": "64",
          "width": "64",
        }
      }>
      <svg
        className="social-icon social-icon--twitter share__icon"
        height="64"
        viewBox="0 0 64 64"
        width="64">
        <g>
          <rect
            fill="#00aced"
            height="64"
            style={Object {}}
            width="64" />
        </g>
        <g>
          <path
            d="M48,22.1c-1.2,0.5-2.4,0.9-3.8,1c1.4-0.8,2.4-2.1,2.9-3.6c-1.3,0.8-2.7,1.3-4.2,1.6 C41.7,19.8,40,19,38.2,19c-3.6,0-6.6,2.9-6.6,6.6c0,0.5,0.1,1,0.2,1.5c-5.5-0.3-10.3-2.9-13.5-6.9c-0.6,1-0.9,2.1-0.9,3.3 c0,2.3,1.2,4.3,2.9,5.5c-1.1,0-2.1-0.3-3-0.8c0,0,0,0.1,0,0.1c0,3.2,2.3,5.8,5.3,6.4c-0.6,0.1-1.1,0.2-1.7,0.2c-0.4,0-0.8,0-1.2-0.1 c0.8,2.6,3.3,4.5,6.1,4.6c-2.2,1.8-5.1,2.8-8.2,2.8c-0.5,0-1.1,0-1.6-0.1c2.9,1.9,6.4,2.9,10.1,2.9c12.1,0,18.7-10,18.7-18.7 c0-0.3,0-0.6,0-0.8C46,24.5,47.1,23.4,48,22.1z"
            fill="white" />
        </g>
      </svg>
    </div>
  </div>
</div>
`;
